capstone/src/main/java/com/google/sps/servlets/SuggestCommentServlet.java

```diff
@@ -0,0 +1,103 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import java.io.BufferedReader;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.sps.data.FeedbackInput;
+import org.json.simple.JSONObject;    
+import org.json.simple.JSONArray;  
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Collection;
+
+/** Servlet that suggests a difference score for Perspective. */
+@WebServlet("/suggest_perspective")
+public class SuggestCommentServlet extends HttpServlet {
+  private static final String URL = "https://commentanalyzer.googleapis.com/v1alpha1/comments:suggestscore?key=API_KEY";
+  private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input
+    FeedbackInput info = new FeedbackInput("", "", new String[0], new String[0]);
+    if (request.getReader() != null) {
+      String data = request.getReader().readLine();
+      Gson gson = new Gson();
+      info = gson.fromJson(data, FeedbackInput.class);
+    }
+    if (info == null) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    String text = info.getText();
+    String lang = info.getLang();
+    HashMap<String, String> feedback = info.getFeedback();
+    
+    // Make the request to Perspective API
+    OkHttpClient client = new OkHttpClient();
+		String json = makePerspectiveJson(text, lang, feedback);
+    String output = post(URL, json, client);
+  
+    // Return Perspective's results
+    response.setContentType("application/json");
+    response.getWriter().println(output);
+  }
+ 
+ 
+  /** Builds the JSON for the body of the call to Perspective API. */
+  private String makePerspectiveJson(String text, String lang, HashMap<String, String> feedback) {
+    JSONObject json = new JSONObject();  
+    JSONObject commentValue = new JSONObject();
+    JSONObject attributesValue = new JSONObject();
+
+    commentValue.put("text", text);
+    for (String suggestion : feedback.keySet()) {
+      JSONObject value = new JSONObject();
+      value.put("value", feedback.get(suggestion));
+      JSONObject summaryScoreValue = new JSONObject();
+      summaryScoreValue.put("summaryScore", value);
+      attributesValue.put(suggestion, summaryScoreValue);
+    }
+
```

Comment:
I think this function could use a comment or two describing how the JSON should be structured, or possibly a general outline of the object we're trying to build (`{ text : __, comment: ___, ... }`). As the code is, it's slightly hard to understand where everything fits.

Reply:
Agreed - I added a comment of that form